src/combineReducers.js

```diff
@@ -74,6 +74,52 @@ function assertReducerSanity(reducers) {
   })
 }
 
+function getFinalReducers(reducers) {
+  var functionReducers = {}
+
+  Object.keys(reducers).map(key => {
+    if (typeof reducers[key] === 'function') {
+      functionReducers[key] = reducers[key]
+    }
+  })
+
+  return functionReducers
+}
+
+function checkStateChange(keys, finalReducers, state, action) {
+  var hasChanged = false
+  var nextState = {}
+  keys.map(key => {
+    var reducer = finalReducers[key]
+    var previousStateForKey = state[key]
+    var nextStateForKey = reducer(previousStateForKey, action)
+
+    if (typeof nextStateForKey === 'undefined') {
+      var errorMessage = getUndefinedStateErrorMessage(key, action)
+      throw new Error(errorMessage)
+    }
+
+    nextState[key] = nextStateForKey
+    hasChanged = hasChanged || nextStateForKey !== previousStateForKey
+  })
+  return hasChanged ? nextState : state
+}
+
+function checkSanityError(reducers) {
+  try {
+    assertReducerSanity(getFinalReducers(reducers))
+  } catch (error) {
+    throw error
```

Comment:
`catch` that rethrows immediately is as good as no `catch`. The error was rethrown later inside the generated reducer deliberately so that Redux DevTools can catch it. We should have added a test verifying this—our fault. But this is wrong and defeats the purpose of having that `try` block.